src/maybe.js

```diff
@@ -14,35 +14,23 @@ class Maybe {
   }
 
   map(fn) {
-    if (this.isNothing()) {
-      return Maybe.of(null);
-    }
-    return Maybe.of(fn(this.value));
+    return Maybe.of(this.isNothing() ? null : fn(this.value));
```

Comment:
Prefer the previous version. It's much easier to read.

Reply:
Actually, this is cool

Reply:
Okay, I could propose a third alternative:

```js
const newValue = this.isNothing() ? null : fn(this.value);
return Maybe.of(newValue);
```

What do you think?

Reply:
Third alternative is great.